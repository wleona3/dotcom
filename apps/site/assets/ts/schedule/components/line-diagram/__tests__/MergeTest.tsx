import React from "react";
import { LineDiagramStop } from "../../__schedule";
import simpleLineDiagram from "./lineDiagramData/simple.json"; // not a full line diagram
import outwardLineDiagram from "./lineDiagramData/outward.json"; // not a full line diagram
import Merges from "../graphics/Merge";
import { mockedRenderWithStopPositionContext } from "../../../../__tests__/test-helpers";
const lineDiagram = (simpleLineDiagram as unknown) as LineDiagramStop[];
const lineDiagramWithBranching = (outwardLineDiagram as unknown) as LineDiagramStop[];

// mock the redux state so that snapshot has positioned stops
const mockState = [...lineDiagram, ...lineDiagramWithBranching].reduce(
  (acc, stop, index) => ({
    ...acc,
    [stop.route_stop.id]: [10, index * 20 + 30]
  }),
  {}
);

describe("Merge component", () => {
  let asFragment: () => DocumentFragment, container: HTMLElement;
  beforeAll(() => {
    ({ asFragment, container } = mockedRenderWithStopPositionContext(
      <svg>
        <Merges lineDiagram={lineDiagramWithBranching} />
      </svg>,
      lineDiagramWithBranching,
      mockState
    ));
  });

  it("renders and matches snapshot", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("shows an SVG group for the merge point", () => {
    expect(container.querySelector("g.line-diagram-svg__merge")).toBeTruthy();
    expect(
      container.querySelector(
        "g.line-diagram-svg__merge line.line-diagram-svg__line"
      )
    ).toBeTruthy();
    expect(
      container.querySelector("g.line-diagram-svg__merge path")
    ).toBeTruthy();
  });

  it("shows nothing when there are no branches", () => {
    const {
      container: containerNoBranches
    } = mockedRenderWithStopPositionContext(
      <svg>
        <Merges lineDiagram={lineDiagram} />
      </svg>,
      lineDiagram,
      mockState
    );

    expect(
      containerNoBranches.querySelector("g.line-diagram-svg__merge")
    ).not.toBeTruthy();
    expect(
      containerNoBranches.querySelector(
        "g.line-diagram-svg__merge line.line-diagram-svg__line"
      )
    ).not.toBeTruthy();
    expect(
      containerNoBranches.querySelector("g.line-diagram-svg__merge path")
    ).not.toBeTruthy();
  });
});
