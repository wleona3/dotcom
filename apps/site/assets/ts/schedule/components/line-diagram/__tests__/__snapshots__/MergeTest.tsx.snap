// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Merge component renders and matches snapshot 1`] = `
<DocumentFragment>
  <svg>
    <g
      class="line-diagram-svg__merge"
    >
      <line
        class="line-diagram-svg__line"
        stroke-width="10px"
        x1="10px"
        x2="10px"
        y1="150px"
        y2="290px"
      />
      <path
        d="M10,250 v-34 a-6,-6 0 0 0 -6,-6 h6"
        stroke-width="10px"
      />
      <path
        d="M10,170 v46 a-6,-6 0 0 0 -6,-6 h6"
        stroke-width="10px"
      />
    </g>
  </svg>
</DocumentFragment>
`;
